### ISSUE TITLE
Select triggers error after retrieving values from server due to 'optionsCaption'

### ISSUE THREAD

**Issue by eko3alpha** [Knockout-Contrib/Knockout-Validation#584]
I want to have a select option that has "Please Select School" as the optionsCaption value.  I'm  loading the values from an external source via ajax once the page loads.

I'm running into 2 frustrating scenarios.

A) I delay the initialization of the validation until AFTER I fetch the select values. This will prevent the validation from triggering an error message because the validation was initialized AFTER the values were loaded/changed. Doing this wont trigger the error message, however I can't trigger the error unless I choose a valid select option and then pick "Please Select School" which has an undefined value that will trigger the error.  Therefore I can click submit without any way to trigger the error on the select.

Or this approach:

B) Initialize the validation first and then update the select values from an external source. However this will trigger the validation error since the select values have been loaded/updated.  This is where my main issue comes into play.  This results in showing the validation error the moment the select options are loaded.

Essentially what I'm asking for is this. Is there a way to manually trigger the form validation and have it display the errors?  I understand that this is a validation library that updates when the observables change, however this is presenting an issue when the initial default value IS an undefined value but you don't want to show an error.

Come to think of it, this would be an issue with a blank input field.  On load its empty and if you try to submit the form I can't find a way to trigger the validation manually.

**Comment by crissdev** [comment]
@eko3alpha Before going into this issue, have you checked out the Getting Started example? http://jsfiddle.net/KHFn8/5424/ It shows how to set up a validation group and toggle visibility of error messages.

**Comment by eko3alpha** [comment]
I did not...

Let me take a look and see where I'm going wrong with my implementation before I go any further.

Thanks, I'll update this with my findings.

**Comment by eko3alpha** [comment]
I found the issue.  I was setting the value of the dropdown with an empty string.  I can duplicate my issue if I modify line 32 on your jsfiddle

    subscription: ko.observable().extend({required: true}),

to:

    subscription: ko.observable('').extend({required: true}),

That empty string was the cause of all my problems.  I was about to lose my mind.  Removing the empty string solved all my issues.  Now the validation library works as I imagined!  I can't believe I spent so much time on this...

Thanks for the link.  It helped
